### PR TITLE
webhook_features: Parse commits with URL with regex

### PR DESCRIPTION
**Pull request by aureleoules**
Fixes https://github.com/MarcoFalke/DrahtBot/issues/16#issuecomment-1326758273.
I think it's better to use regex for this anyway as it's more testable and reliable.

### CODE REVIEW

File: webhook_features/src/features/summary_comment.rs [hunk 1]
```diff
@@ -296,7 +296,10 @@ lazy_static! {
         (r"\b(Approach NACK)\b", AckType::ApproachNack),
         (r"\b(NACK)\b", AckType::ConceptNack),
         (r"\b(Concept ACK)\b", AckType::ConceptAck),
-        (r"(ACK)(?:\s*)([0-9a-f]{6,40})\b", AckType::Ack),
+        (
+            r"(ACK)(?:\s*)(((https://github\.com/.*/commits?/)|\[)?([0-9a-f]{6,40}))\b",
```
Comment (MarcoFalke) [inline webhook_features/src/features/summary_comment.rs:300]:
```suggestion
            r"(ACK)(?:.*?)([0-9a-f]{6,40})\b",
```

Comment (MarcoFalke) [reply webhook_features/src/features/summary_comment.rs:300]: nit: What about just taking the first commit hash that follows?

**Review by MarcoFalke — APPROVED** [review]

File: webhook_features/src/features/summary_comment.rs (hunk 1 above)
Comment (aureleoules) [reply webhook_features/src/features/summary_comment.rs:300]: I'm a bit afraid of comments like "I would love to ACK but I don't understand the code in `1235678`".

Comment (MarcoFalke) [reply webhook_features/src/features/summary_comment.rs:300]: Well, it is never possible to be free of false-positives. There is also "Almost ACK 1235678, if the bug is fixed"

Comment (aureleoules) [reply webhook_features/src/features/summary_comment.rs]: Sure, pushed.

**Comment by aureleoules** [comment]
Rebased

**Pull request merged**